File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    max_count = 0
    mod_num = 0
    counter = set(array)
    for c in counter:
        if max_count < array.count(c):
            max_count = array.count(c)
            mod_num = c
        elif max_count == array.count(c) and mod_num != c:
            mod_num = -1
    return mod_num

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;
class Solution {
    public int solution(int[] array) {
        Arrays.sort(array);
        int answer = 0;
        int count = 0;
        int tmpc = -1;
        int tmpn = -1;
        for (int n : array){
            if (n != tmpn){
                if(tmpc > count){
                    answer = tmpn;
                    count = tmpc;
                }
                else if(tmpc == count){
                    answer = -1;
                }
                tmpc = 1;
                tmpn = n;
            }else{
                tmpc += 1;
            }
        }
        if(tmpc > count){
            answer = tmpn;
            count = tmpc;
        }
        else if(tmpc == count){
            answer = -1;
        }
        return answer;
    }
}